lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -5886,6 +5886,32 @@ class DecomposeAtenCosineSimilarityOp
 };
 } // namespace
 
+namespace {
+// decompose `trunc(x)` to `sign(x) * floor(abs(x))`
+class DecomposeAtenTruncOp : public OpRewritePattern<AtenTruncOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTruncOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.getSelf();
+
+    auto resultTy = dyn_cast<ValueTensorType>(op.getType());
+    if (!resultTy || !resultTy.hasDtype()) {
+      return rewriter.notifyMatchFailure(op, "result must have dtype");
+    }
+
+    if (resultTy.getDtype().isa<mlir::FloatType>()) {
```

Comment:
Could you please modify functional style `isa<mlir::FloatType>(resultTy.getDtype())`
We should prefer functional style as the method style is deprecated https://github.com/llvm/mlir-www/blob/main/website/content/deprecation/_index.md#deprecated

Reply:
Done